File: csharp/space-age/SpaceAge.cs
using System;

public class SpaceAge
{
    private const double SecondsInEarthYear = 31557600d;

    private const double MercuryOrbitRatio = 0.2408467;
    private const double VenusOrbitRatio = 0.61519726;
    private const double MarsOrbitRatio = 1.8808158;
    private const double JupiterOrbitRatio = 11.862615;
    private const double SaturnOrbitRatio = 29.447498;
    private const double UranusOrbitRatio = 84.016846;
    private const double NeptuneOrbitRatio = 164.79132;

    private readonly int seconds;

    public SpaceAge(int seconds) => this.seconds = seconds;

    public double OnEarth() => seconds / SecondsInEarthYear;

    public double OnMercury() => OnEarth() / MercuryOrbitRatio;

    public double OnVenus() => OnEarth() / VenusOrbitRatio;

    public double OnMars() => OnEarth() / MarsOrbitRatio;

    public double OnJupiter() => OnEarth() / JupiterOrbitRatio;

    public double OnSaturn() => OnEarth() / SaturnOrbitRatio;

    public double OnUranus() => OnEarth() / UranusOrbitRatio;

    public double OnNeptune() => OnEarth() / NeptuneOrbitRatio;
}

File: java/space-age/src/main/java/SpaceAge.java
import java.util.HashMap;

class SpaceAge {

    final static double earthSeconds = 31557600d;

    static final HashMap<String, Double> ages;
    static{
        ages = new HashMap<>();
        ages.put("Earth"  ,               earthSeconds);
        ages.put("Mercury", 0.2408467   * earthSeconds);
        ages.put("Venus"  , 0.61519726  * earthSeconds);
        ages.put("Mars"   , 1.8808158   * earthSeconds);
        ages.put("Jupiter", 11.862615   * earthSeconds);
        ages.put("Saturn" , 29.447498   * earthSeconds);
        ages.put("Uranus" , 84.016846   * earthSeconds);
        ages.put("Neptune", 164.79132   * earthSeconds);
    }

    double seconds;

    SpaceAge(double seconds) {
        this.seconds = seconds;
    }

    double onEarth() {
        return seconds / ages.get("Earth");
    }

    double onMercury() {
        return seconds / ages.get("Mercury");
    }

    double onVenus() {
        return seconds / ages.get("Venus");
    }

    double onMars() {
        return seconds / ages.get("Mars");
    }

    double onJupiter() {
        return seconds / ages.get("Jupiter");
    }

    double onSaturn() {
        return seconds / ages.get("Saturn");
    }

    double onUranus() {
        return seconds / ages.get("Uranus");
    }

    double onNeptune() {
        return seconds / ages.get("Neptune");
    }

}
